calyx/src/analysis/live_range_analysis.rs

```diff
@@ -612,8 +612,32 @@ impl LiveRangeAnalysis {
         invoke: &ir::Invoke,
         shareable_components: &ShareSet,
     ) -> (TypeNameSet, TypeNameSet) {
-        //The reads of the invoke include its inputs plus the cell itself, if the
-        //outputs are not empty.
+        let mut comp_is_written = false;
+
+        //The writes of the invoke include its outpus plus the cell itself, if the
+        //inputs are not empty.
+        let mut write_set: TypeNameSet = invoke
+            .outputs
+            .iter()
+            .filter_map(|(_, src)| {
+                Self::port_to_cell_name(src, shareable_components)
+            })
+            .collect();
+        if !invoke.inputs.is_empty()
```

Comment:
To avoid another mutable variable, maybe just

```rust
let comp_is_written =  !invoke.inputs.is_empty() && shareable_components.is_shareable_component(&invoke.comp);
if (comp_is_written)
{
 write_set.insert((
    invoke.comp.borrow().prototype.clone(),
    invoke.comp.borrow().clone_name(),
  ));
}
```